plugins/thunderbolt/fu-plugin-thunderbolt.c

```diff
@@ -744,6 +756,37 @@ fu_plugin_update (FuPlugin *plugin,
 		return FALSE;
 	}
 
+	validation = fu_plugin_thunderbolt_validate_firmware (udevice,
+							      blob_fw,
+							      &error_local);
+	if (validation != VALIDATION_PASSED) {
+		g_autofree gchar* msg = NULL;
+		switch (validation) {
+		case VALIDATION_FAILED:
+			msg = g_strdup_printf ("could not validate firmware: %s",
+					       error_local->message);
+			break;
+		case VALIDATION_PASSED_UNKNOWN_DEVICE:
+			msg = g_strdup ("firmware validation seems to be passed but the device is unknown");
+			break;
+		default:
+			break;
+		}
+		if (!force) {
+			g_set_error (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INVALID_FILE,
+				     "%s\n"
```

Comment:
This looks odd: maybe just have `g_set_error_literal` with just the msg. I don't think it's a good idea we actually suggest to the user to use `--force` in an error message.

Reply:
That is how things are done in the Dell plugin too currently.
https://github.com/hughsie/fwupd/blob/master/plugins/dell/fu-plugin-dell.c#L732

If you'd like to prevent telling anyone about --force, maybe these types of errors should be redirected to a wiki page instead.

Reply:
(which is exactly what I used as a reference :) )

Reply:
I don't think we should be telling the user to do something risky; updating firmware is supposed to be "safe" with fwupd. I think it's fine to reference a wiki page explaining the issues in more depth.

Reply:
Adjusted Dell plugin for this.
https://github.com/hughsie/fwupd/commit/1790115c933f57670ec5250ce6d4d5faa0987293
Feel free to create a similar one for TBT

Reply:
Thanks, @superm1!
I want an internal review in my team for the wiki page I wrote before uploading it, so for now I'm just removing the unsafe suggestion. When I have the wiki page uploaded, I'll update the message here in lines with what you did.